kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/JobOperationsImpl.java

```diff
@@ -223,4 +223,24 @@ public LogWatch watchLog(OutputStream out) {
     }
     return null;
   }
+
+  @Override
+  public Job replace(Job job) {
+    if (job == null) {
+      job = getItem();
+    }
+    // Fetch item from server and patch Selector and PodTemplate
+    // metadata in case not present already in order to avoid 422
+    Job jobFromServer = fromServer().get();
+    if (job.getSpec().getSelector() == null) {
+      job.getSpec().setSelector(jobFromServer.getSpec().getSelector());
+    }
+    if (job.getSpec().getTemplate().getMetadata() != null) {
+      job.getSpec().getTemplate().getMetadata().setLabels(jobFromServer.getSpec().getTemplate().getMetadata().getLabels());
```

Comment:
Is it possible that the replacing job (new) will contain additional labels?
If so, shouldn't we purAll labels from server and keep the current. 

What if replacing job is removing some labels, is this the expected behavior?

Reply:
`Job` resource's labels are different from `PodTemplate`'s labels. I think these are assigned from kube-apiserver after initial creation.